nhais-adaptor/edifact/models/message.py

```diff
@@ -0,0 +1,161 @@
+        segments = [
+            Segment(key="S01", value="1"),
+            Segment(key="RFF", value=f"TN:{transaction_number}"),
+            Segment(key="NAD", value=f"GP+{party_id}:900"),
+            Segment(key="HEA", value=f"ACD+{acceptance_code}:ZZZ"),
+            Segment(key="HEA", value=f"ATP+{acceptance_type}:ZZZ"),
+            Segment(key="DTM", value=f"956:{formatted_date_time}:102"),
+            Segment(key="LOC", value=f"950+{location}"),
+        ]
+        super().__init__(segments=segments)
+
+
+class MessageSegmentTrigger2(SegmentCollection):
+    """
+    A collection of segments that represent personal information about a patient.
+    """
+
+    def __init__(self, id_number, name, date_of_birth, gender, address):
+        """
+        :param id_number: OPI official Payment Id (existing NHS Number)
+        :param name: the name of the patient
+        :param date_of_birth: Patients date of birth
+        :param gender: sex of the patient. For an acceptance transaction, reference "G1", this segment is required
+        :param address: the patients address
+        """
+        formatted_date = datetime.strptime(date_of_birth, '%Y-%m-%d').strftime('%Y%m%d')
+        segments = [
+            Segment(key="S02", value="2"),
+            Segment(key="PNA",
+                    value=f"PAT+{id_number}:OPI+++SU:{name.family_name}+FO:{name.first_given_forename}+TI:{name.title}+MI:{name.middle_name}+FS:{name.third_given_forename}"),
+            Segment(key="DTM", value=f"329:{formatted_date}:102"),
+            Segment(key="PDI", value=f"{gender}"),
+            Segment(key="NAD",
+                    value=f"PAT++{address.house_name}:{address.address_line_1}:{address.address_line_2}:{address.town}:{address.county}+++++{address.post_code}")
+        ]
+        super().__init__(segments=segments)
+
+
+class Message(SegmentCollection):
+    """
+    An edifact Message that is contained within an interchange
+    a collection of Segments
+    """
+
+    def __init__(self, sequence_number, message_beginning, message_segment_trigger_1, message_segment_trigger_2):
+        """
+        :param sequence_number: the unique sequence number of the message
+        :param message_beginning: the beginning of the message
+        :param message_segment_trigger_1: Segment trigger 1 registration information
+        :param message_segment_trigger_2: Segment trigger 2 personal information about patient
+        """
+        msg_header = MessageHeader(sequence_number=sequence_number)
+        number_of_segments = message_beginning.size() + message_segment_trigger_1.size() + message_segment_trigger_2.size() + 2
+        msg_trailer = MessageTrailer(number_of_segments=number_of_segments, sequence_number=sequence_number)
+        segments = [msg_header, message_beginning, message_segment_trigger_1, message_segment_trigger_2, msg_trailer]
+        super().__init__(segments=segments)
+
+
+class Messages(object):

```

Comment:
Would it hold any value to have these collection classes inherit from the list class? This would allow you to append messages instead of passing a list in and wouldn't need a custom size method.

Reply:
👍